torchx/specs/api.py

```diff
@@ -189,8 +190,25 @@ def apply(self, role: "Role") -> "Role":
             role = copy.deepcopy(role)
             role.args = [self.substitute(arg) for arg in role.args]
             role.env = {key: self.substitute(arg) for key, arg in role.env.items()}
+            role.metadata = self._apply_nested(role.metadata)
```

Comment:
could you add a test-case for this?

Reply:
There is a test case on the meta side of this make me can also add a version of it to api test @jason-b-akers

Reply:
Hi @kiukchung. There's an internal test in [D58678835](https://www.internalfb.com/diff/D58678835). As a follow-up I'll write one in torchx/schedulers/test/api_test.py as well.

Reply:
Follow-up test in [PR#293](github.com/pytorch/torchx/pull/923)